src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -152,8 +210,43 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
           message: t('Field is required'),
         });
       });
+
+      // Check thresholds
+      this.isValidTrigger(
+        changedTriggerIndex ?? triggerIndex,
+        trigger,
+        triggerErrors,
+        changeObj
+      );
     });
 
+    // If we have 2 triggers, we need to make sure that the critical and warning
+    // alert thresholds are valid (e.g. if critical is above x, warning must be less than x)
+    if (triggers.length === 2) {
+      const criticalTriggerIndex = triggers.findIndex(({label}) => label === 'critical');
+      const warningTriggerIndex = criticalTriggerIndex ^ 1;
```

Comment:
Please what you want to achieve here. I've never used `^ ` before?

Reply:
`^` is XOR, we know there are two array elements with indexes 0 and 1, given a single index, we're looking for the other one.